TypeCobol.Analysis/Cfg/ControlFlowGraphBuilder.cs

```diff
@@ -0,0 +1,2695 @@
+        private IEnumerable<CfgSentence> YieldSectionOrParagraphSentences(Symbol sectionOrParagraphSymbol)
+        {
+            if (sectionOrParagraphSymbol != null)
+            {
+                if (sectionOrParagraphSymbol.Kind == Symbol.Kinds.Paragraph)
+                {
+                    CfgParagraphSymbol cfgParaSymbol = (CfgParagraphSymbol)sectionOrParagraphSymbol;
+                    foreach (var sentence in cfgParaSymbol.Sentences)
+                    {
+                        yield return sentence;
+                    }
+                }
+                else
+                {//This is a section.  
+                    CfgSectionSymbol cfgSection = (CfgSectionSymbol)sectionOrParagraphSymbol;
+                    foreach (var part in cfgSection.SentencesParagraphs)
+                    {
+                        System.Diagnostics.Debug.Assert(part.Kind == Symbol.Kinds.Sentence || part.Kind == Symbol.Kinds.Paragraph);
+                        if (part.Kind == Symbol.Kinds.Paragraph)
+                        {
+                            CfgParagraphSymbol cfgParaSymbol = (CfgParagraphSymbol)part;
+                            foreach (var sentence in cfgParaSymbol.Sentences)
+                            {
+                                yield return sentence;
+                            }
+                        }
+                        else
+                        {
+                            CfgSentence sentence = (CfgSentence)part;
+                            yield return sentence;
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Resolve a Goto from a block
+        /// </summary>
+        /// <param name="goto">The target goto</param>
+        /// <param name="block">The source block</param>
+        /// <param name="target">The target sections or paragraphs
+        /// </param>
+        /// <param name="v">True for a simple Goto</param>
+        /// <returns>true if all targets have been resolved, false otherwise.</returns>
+        private bool ResolveGoto(Goto @goto, BasicBlockForNode block, SymbolReference[] target, bool simpleGoto)
+        {
+            HashSet<Symbol> targetSymbols = new HashSet<Symbol>();
+            foreach (var sref in target)
+            {
+                bool bHasOne = false;
+                Symbol targetSymbol = null;
+                IEnumerable<CfgSentence> sentences = ResolveSectionOrParagraphSentences(@goto, sref, out targetSymbol);
+                if (targetSymbol == null)
+                {
+                    Diagnostic d = new Diagnostic(MessageCode.SemanticTCErrorInParser,
+                        @goto.CodeElement.Column,
+                        @goto.CodeElement.Column,
+                        @goto.CodeElement.Line,
+                        string.Format(Resource.UnknownSectionOrParagraph, sref.ToString()));
+                    Diagnostics.Add(d);
```

Comment:
Duplicate diagnostic with the one directly created in `CheckSectionOrParagraph` method.

Reply:
ok

Reply:
Diagnostic has been deleted.